initfiles/bin/init_thorslave.in

```diff
@@ -44,24 +44,14 @@ slavename=thorslave_${hpcc_compname}
 stop_slaves()
 {
     local timer=15
-    local isAlive=0
-
-    log "Attempting to kill $slavename with SIGTERM"
-    killall -SIGTERM $slavename > /dev/null 2>&1
-    while [[ $isAlive -eq 0 && $timer -gt 0 ]];do
-        killall -0 $slavename > /dev/null 2>&1
-        isAlive=$?
-        [[ $isAlive -eq 0 ]] && sleep 0.5
-        ((timer--))
+    local _pidarray
+    local _pidname
+    
+    _pidarray=($( ls -l ${PID_DIR} | awk -v regpattern=${slavename}_[1-9][0-9]*\.pid '$0~regpattern {print $9;}' ))
+    for _pidname in ${_pidarray[@]}; do
+        kill_process ${PID_DIR}/$_pidname $slavename $timer
     done
-
-    if [[ $isAlive -eq 0 ]]; then
```

Comment:
pidwait_fn is called with the Force flag from within kill_process. This logic is duplicated with kill inside that function.